project/Dependencies.scala

```diff
@@ -125,7 +127,7 @@ object Dependencies {
 
   lazy val httpJackson = l ++= Seq(jackson)
 
-  lazy val docs = l ++= Seq(Docs.sprayJson, Docs.gson, Docs.jacksonXml, Docs.reflections)
+  lazy val docs = l ++= Seq(Docs.sprayJson, Docs.gson, Docs.jacksonXml, Docs.reflections, Docs.cats, Docs.scalaz)
```

Comment:
I'm not sure we want to show cats/scalaz in our documentation - people who are using that probably already know how to create such typeclasses?

Reply:
Should I remove Scalaz and Cats examples from `HttpRequestShow` and `HttpResponseShow`. I kind of like the idea of having both options explained:

Some Cats and Scalaz users are unaware of the `Show[T]` typeclass.

Non-users might be completely unaware of the fact that these libraries exist or that they provide typeclasses and syntax to add to-string methods on top of existing classes and think: _Oh, well now I have to write a trait, and object, implicit instances and implicit conversions to have what I used to have_.

Reply:
Not sure - I can see how it might give the reader an interesting pointer, on the other hand it increases the scope of the docs making them harder to understand and to maintain. Perhaps another reviewer can weigh in?

> now I have to write a trait, and object, implicit instances and implicit conversions to have what I used to have

I'm not sure they'd have to go that full mile - they could just write a function?

Reply:
I've no problem with removing the Cats and Scalaz examples and adding simpler ones (using vanilla Scala) + keeping the  from-scratch type class.

Reply:
This commit https://github.com/akka/akka-http/pull/2560/commits/409bbc98044e0bcbaa84f6b7cb61a2eb6389958b 

Removes Cats and Scalaz dependencies and the examples using them. I initially included these because I liked @jypma suggestion in https://github.com/akka/akka-http/pull/2412#issuecomment-466986505 . 

But I also understand that you don't want to introduce additional dependencies.

In any case, this is done in the last commit so if you finally decide that you want to include the examples, let me know and I can drop that commit.

Also there is a commit before that one adding examples using extension classes.